src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,6 +37,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        String[] splitaddress = trimmedAddress.split(DELIMITER);
+        this.block = new Block(splitaddress[0]);
```

Comment:
why not 1? or 2? or 100?
I hope you understand what I mean! ;-) may be i revoke the good from the previous comment :-P